app/js/common/header.jsx

```diff
@@ -8,97 +8,162 @@
  * graphic logo is a trademark of OpenMRS Inc.
  */
 
-import React, {Component} from 'react';
-import {Link, IndexLink} from 'react-router';
+import React, { Component } from 'react';
+import { Link, IndexLink } from 'react-router';
 import apiCall from '../utilities/apiHelper';
 
-/**
- * Represents the menu bar.
- * 
- * @class Header
- * @extends {Component}
- */
 export default class Header extends Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     this.state = {
       locationTags: [],
       currentLocationTag: "",
+      currentLocation: "",
       currentUser: "",
       currentLogOutUrl: "",
+      contextPath: "",
+      columns: 0,
+      selected: "",
     };
     this.getUri = this.getUri.bind(this);
+    this.handleLocationChange = this.handleLocationChange.bind(this);
+    this.locationsDropDown = this.locationsDropDown.bind(this);
   }
 
-  /**
-   * Gets the OpenMrs configuration and sets the logout link
-   * @memberOf Header
-   */
   componentWillMount() {
-    apiCall(null, 'get', '/location').then((response) => {
-      this.setState({locationTags: response.results});
+    apiCall(null, 'get', '/location?tag=Login%20Location').then((response) => {
+      this.setState({ locationTags: response.results },
+        () => {
+          const columns = Math.floor(this.state.locationTags.length / 3);
+          this.setState({
+            columns
+          });
+        }
+      );
       this.getUri();
     });
 
-    apiCall(null, 'get', '/session').then((response) => {
-      this.setState({currentUser: response.user.display});
+    apiCall(null, 'get', 'appui/session?v=full').then((response) => {
+      this.setState({
```

Comment:
you could destructure here and get what you need from the res as const like eg
`const {currentUser , currentLocation etc } = response`

Reply:
@CeciliaWanjiku  I see what you mean.  However destructure will not apply for above as the name from response is different.

Reply:
@CeciliaWanjiku  on rechecking, it was possible.

Resolved.